massa-storage/src/endorsement_indexes.rs

```diff
@@ -40,28 +46,47 @@ impl EndorsementIndexes {
         }
     }
 
+    /// Link a vector of endorsements to a block. Should be used in case of the block is added to the storage.
+    /// Arguments:
+    /// - block_id: the block id to link the endorsements to
+    /// - endorsement_ids: the endorsements to link to the block
     pub(crate) fn link_endorsements_with_block(
         &mut self,
         block_id: &BlockId,
-        operations: &Vec<EndorsementId>,
+        endorsement_ids: &Vec<EndorsementId>,
     ) {
         self.index_by_block
             .entry(*block_id)
             .or_default()
```

Comment:
This won't insert the element if it is missing.

Use `.or_insert_with(|| Default::default())`